tests/bundle/bundle-test.ts

```diff
@@ -129,214 +29,344 @@ async function runAll() {
-    console.info('DID successfully deleted')
-  } else {
-    throw new Error('DID was not deleted')
-  }
+  // ┏━━━━━━━━━━━━━━━━┓
+  // ┃ Add a service  ┃
+  // ┗━━━━━━━━━━━━━━━━┛
+  const addServiceTransactionResult = await Kilt.DidHelpers.addService({
+    api,
+    submitter,
+    signers,
+    didDocument,
+    // TODO:  change service endpoint.
+    service: {
+      id: '#my_service',
+      type: ['http://schema.org/EmailService'],
+      serviceEndpoint: ['mailto:info@kilt.io'],
+    },
+  }).submit()
 
-  // CType workflow
-  console.log('CType workflow started')
+  if (addServiceTransactionResult.status !== 'confirmed') {
+    throw new Error('add service failed')
+  }
+  ;({ didDocument } = addServiceTransactionResult.asConfirmed)
+
+  console.log('service added')
+
+  // ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+  // ┃ Register a CType             ┃
+  // ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+  //
+  // Register a credential type on chain so we can issue credentials against it.
+  //
+  // Note:
+  // We are registering a CType that has been created previously using functionality from the @kiltprotocol/credentials package.
+  // The @kiltprotocol/sdk-js package and bundle do not currently offer support for this.
+  //
+  // TODO: Decide if CType definitions are expected to be hardcoded in application logic, at least for credential issuance.
+  // Verifying credentials / presentations is already possible even if the CType definition is not known.
+  //
   const DriversLicenseDef =
     '{"$schema":"ipfs://bafybeiah66wbkhqbqn7idkostj2iqyan2tstc4tpqt65udlhimd7hcxjyq/","additionalProperties":false,"properties":{"age":{"type":"integer"},"name":{"type":"string"}},"title":"Drivers License","type":"object"}'
 
-  const cTypeStoreTx = await authorizeTx(
+  const createCTypeResult = await Kilt.DidHelpers.transact({
     api,
-    api.tx.ctype.add(DriversLicenseDef),
-    alice.id,
-    aliceSign(alice)[0],
-    payerSigner.id
-  )
-
-  const result = await signAndSubmitTx(cTypeStoreTx, payerSigner)
+    didDocument,
+    signers,
+    submitter,
+    call: api.tx.ctype.add(DriversLicenseDef),
+    expectedEvents: [{ section: 'CType', method: 'CTypeCreated' }],
+  }).submit()
+
+  if (createCTypeResult.status !== 'confirmed') {
+    throw new Error('CType creation failed')
+  }
 
-  const ctypeHash = result.events
-    ?.find((ev) => api.events.ctype.CTypeCreated.is(ev.event))
-    ?.event.data[1].toHex()
+  // TODO: We don't have the CType id in the definition, so we need to get it from the events.
+  const ctypeHash = createCTypeResult.asConfirmed.events
+    .find((event) => api.events.ctype.CTypeCreated.is(event))
+    ?.data[1].toHex()
 
-  if (!ctypeHash || !(await api.query.ctype.ctypes(ctypeHash)).isSome) {
-    throw new Error('storing ctype failed')
+  if ((await api.query.ctype.ctypes(ctypeHash)).isEmpty) {
+    throw new Error('CType not registered')
   }
 
+  // TODO: Should we at least be able to load an existing CType from the chain?
```

Comment:
@rflechtner why is this not possible?

Reply:
You can do so via importing the CTypeLoader from `@kiltprotocol/credentials`. But the bundle only has sdk-js functions available.